collector/transport_node_collector_test.go

```diff
@@ -0,0 +1,318 @@
+package collector
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+func fakeTransportNodeID(id string) string {
+	return fmt.Sprintf("fake-transport-node-id-%s", id)
+}
+
+func fakeTransportNodeName(id string) string {
+	return fmt.Sprintf("fake-transport-node-name-%s", id)
+}
+
+func fakeTransportZoneID(id string) string {
+	return fmt.Sprintf("fake-transport-zone-id-%s", id)
+}
+
+func fakeEdgeClusterID(id string) string {
+	return fmt.Sprintf("fake-edge-cluster-id-%s", id)
+}
+
+type transportNodeStatusResponse struct {
+	ID     string
+	Status string
+	Error  error
+}
+
+type transportNodeClientMock struct {
+	edgeClustersResponse         []manager.EdgeCluster
+	edgeClustersError            error
+	transportNodeStatusResponses []transportNodeStatusResponse
+}
+
+func (c *transportNodeClientMock) ListAllTransportNodes() ([]manager.TransportNode, error) {
```

Comment:
codes that are calling `ListAllTransportNodes` cannot be tested because it's in `Collect()` only?

Reply:
Yes, it's not being called